### PR TITLE
exit w/ non-zero result when any tests fail

### PR DESCRIPTION
**Pull request by heckj**
offering up as a fix for #10

### CODE REVIEW

File: Sources/swift-doctest/main.swift [hunk 1]
```diff
@@ -81,6 +90,22 @@ struct SwiftDocTest: ParsableCommand {
...
         standardOutput.write(consolidatedReport.description.data(using: .utf8)!)
+        let failureList = consolidatedReport.results.filter({ result -> Bool in
+            do {
+                return try !result.get().ok
+            } catch {
+
+                return true
+            }
+        })
+        if (failureList.count > 0) {
```
Comment (mattt) [inline Sources/swift-doctest/main.swift]:
```suggestion
      if consolidatedReport.results.contains(where: { (try? $0.get().ok) != true }) {
```

File: Sources/swift-doctest/main.swift [hunk 2]
```diff
@@ -81,6 +90,22 @@ struct SwiftDocTest: ParsableCommand {
...
+            // return a non-zero result code so that CI systems will react appropriately
+            #if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
+            Darwin.exit(Int32(failureList.count))
+            #elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android) || os(Cygwin) || os(Haiku)
+            Glibc.exit(Int32(failureList.count))
+            #endif
+        }
```
Comment (mattt) [inline Sources/swift-doctest/main.swift]:
```suggestion
            // Return a non-zero result code if any tests failed
            #if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
            Darwin.exit(EXIT_FAILURE)
            #elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android) || os(Cygwin) || os(Haiku)
            Glibc.exit(EXIT_FAILURE)
            #endif
        }
```

Comment (mattt) [reply Sources/swift-doctest/main.swift]: I don't think we should signal with an exit status other than success or failure. Passing the number of failed tests has the potential of signaling a status in excess of `255` (which may cause problems of some platforms) and deprives us of the ability to designate status codes for certain kinds of failures.

File: Sources/swift-doctest/main.swift [hunk 3]
```diff
@@ -2,6 +2,15 @@ import ArgumentParser
...
 import TAP
+// pattern borrowed upstream from swift:
+// https://github.com/apple/swift/blob/master/test/Interpreter/SDK/libc.swift#L10-L14
+#if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
+  import Darwin
+#elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android) || os(Cygwin) || os(Haiku)
+  import Glibc
+#elseif os(Windows)
+  import MSVCRT
+#endif
```
Comment (mattt) [inline Sources/swift-doctest/main.swift]:
```suggestion
// Pattern borrowed upstream from Swift: 
// https://github.com/apple/swift/blob/87d3b4d984281b113ffad503cdb1d82b9f0ae5b9/test/Interpreter/SDK/libc.swift#L12-L17
#if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
  import Darwin
#elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android) || os(Cygwin) || os(Haiku)
  import Glibc
#elseif os(Windows)
  import MSVCRT
#endif
```

Comment (mattt) [reply Sources/swift-doctest/main.swift]: Using a permalink ensures that file and line references are stable (when I visited the provided URL, the highlighted portion was off by a couple lines)

File: Sources/swift-doctest/main.swift (hunk 1 above)
Comment (mattt) [reply Sources/swift-doctest/main.swift]: See review comment below. If we don't need the result of `map`, we should use `contains(where:)` instead.

**Review by mattt — CHANGES_REQUESTED** [review]

**Comment by mattt** [comment]
@heckj Exiting with a deliberate status code is absolutely the correct thing to do. Thanks for identifying this and providing a fix. Just a few small notes on my end, whenever you have a chance to take a look.

**Comment by heckj** [comment]
first time I've seen changes requested inline with github's pull requests - that was super interesting. How did you (as the maintainer/repo owner) mechanically do that? Did you add one by one commits, or all inline within a browser?

**Comment by mattt** [comment]
> first time I've seen changes requested inline with github's pull requests - that was super interesting. How did you (as the maintainer/repo owner) mechanically do that? Did you add one by one commits, or all inline within a browser?

All inline from the browser. Super convenient! Decided to tweet that out to spread the word about my favorite new feature on GitHub: https://twitter.com/mattt/status/1252289846493626372

**Review by mattt — APPROVED** [review]

**Comment by mattt** [comment]
Thanks again for your help with this, @heckj! I just released 0.0.3, which includes this change.

**Comment by heckj** [comment]
awesome, thanks for pulling this all together! I'm aiming to use it on at least one project I'm working on now: https://github.com/heckj/OpenTelemetryModels/pull/7

**Comment by mattt** [comment]
@heckj Very cool. Sorry that this isn't working on Linux yet. That's the next thing I want to fix.

**Pull request merged**